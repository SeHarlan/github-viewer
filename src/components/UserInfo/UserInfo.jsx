import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ name, followersCount, followingCount, githubLink }) => (
  <>
    <h1>{name}</h1>
    <p>Followers: {followersCount}</p>
    <p>Following: {followingCount}</p>
    <a href={githubLink}>Github Profile</a>
  </>
);

UserInfo.propTypes = {
  name: PropTypes.string.isRequired,
  followersCount: PropTypes.number.isRequired,
  followingCount: PropTypes.number.isRequired,
  githubLink: PropTypes.string.isRequired
};

export default UserInfo;
