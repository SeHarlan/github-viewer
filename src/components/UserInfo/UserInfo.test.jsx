import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './UserInfo';

describe('UserInfo component', () => {
  it('renders UserInfo', () => {
    const wrapper = shallow(<UserInfo name="Scott" followersCount={2} followingCount={4} githubLink="link.com" />);

    expect(wrapper).toMatchSnapshot();
  });
});
