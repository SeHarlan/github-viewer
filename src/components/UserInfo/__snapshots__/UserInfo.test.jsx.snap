// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component renders UserInfo 1`] = `
<Fragment>
  <h1>
    Scott
  </h1>
  <p>
    Followers: 
    2
  </p>
  <p>
    Following: 
    4
  </p>
  <a
    href="link.com"
  >
    Github Profile
  </a>
</Fragment>
`;
